### PR TITLE
Deployment updates

### PR DESCRIPTION
**Pull request by lemieux**
This adds the new deployment steps with for script loader. I replace grunt stuff with direct calls to the aws cli (it's available on Circle). I also split the umd loader in 3 different builds for convenience (and also, uglifyJS stopped liking our umd.js file for some reason.

tl;dr; when deploying there's an extra step that uploads a json file with the new script url in it in the webloader S3 bucket and then issues an invalidation request to cloudfront.

I tested it all with a whole dummy infra to replicate prod.

### CODE REVIEW

File: make-webpack-config.js [hunk 1]
```diff
@@ -150,7 +153,7 @@ module.exports = function(options) {
         options.publicPath :
         buildType === 'dev' ?
             '/_assets/' :
-            'https://cdn.smooch.io/';
+            'https://cdn.smooch.rocks/';
```
Comment (mspensieri) [inline make-webpack-config.js]: leftover from testing?

File: make-webpack-config.js [hunk 2]
```diff
@@ -169,7 +172,7 @@ module.exports = function(options) {
         filename: baseFilename + fileExtension,
         chunkFilename: buildType === 'dev' ? '[id].js' : '[chunkhash].js',
         sourceMapFilename: '[file].map',
-        library: buildType === 'host' ? 'Smooch' : undefined,
+        library: undefined,
```
Comment (mspensieri) [inline make-webpack-config.js]: ?

File: make-webpack-config.js (hunk 1 above)
Comment (lemieux) [inline make-webpack-config.js]: oh..

File: make-webpack-config.js (hunk 2 above)
Comment (lemieux) [inline make-webpack-config.js]: fixed

Comment (lemieux) [inline make-webpack-config.js]: Webpack doesn't need to expose a global variable anymore, it's handled by the script loader.

**Pull request merged**